MediaStore/app/src/main/java/com/android/samples/mediastore/MainActivityViewModel.kt

```diff
@@ -185,6 +200,48 @@ class MainActivityViewModel(application: Application) : AndroidViewModel(applica
         return images
     }
 
+    private suspend fun performDeleteImage(image: MediaStoreImage) {
+        // Starting a delete clears the previous pending image.
+        pendingDeleteImage = image
```

Comment:
Probably fine, but looks racey. I get the feeling that the pending deletes should be modeled as a collection or `Flow`, since there could be multiple.

Reply:
I updated it a bit, but given that you have to click on a button in a dialog to pick a new image, I don't think it's really necessary to worry about queued up deletes in this case?